src/schema.ts

```diff
@@ -261,31 +272,65 @@ UserSchema.virtual("oldEnoughForWithdrawal").get(function (this: typeof UserSche
   return d - this.created_at.getTime() > yamlConfig.limits.oldEnoughForWithdrawal
 })
 
-UserSchema.methods.limitHit = async function ({
-  on_us,
-  amount,
-}: {
-  on_us: boolean
-  amount: number
-}) {
-  const timestampYesterday = Date.now() - MS_PER_DAY
+UserSchema.virtual("twoFactorEnabled").get(function (this: typeof UserSchema) {
+  return !!this.twoFactor.secret
+})
 
-  const txnType = on_us
-    ? [{ type: "on_us" }, { type: "onchain_on_us" }]
-    : [{ type: { $ne: "on_us" } }]
+const getTimestampYesterday = () => Date.now() - MS_PER_DAY
 
-  const limit = yamlConfig.limits[on_us ? "onUs" : "withdrawal"].level[this.level]
+// FIXME: Improve function name
+UserSchema.methods.remainingTwoFactorLimit = async function () {
+  const threshold = this.twoFactor.threshold
+
+  const txnType = [
+    { type: "on_us" },
+    { type: "onchain_on_us" },
+    { type: "onchain_payment" },
+    { type: "payment" },
+  ]
 
   const outgoingSats =
     (
       await User.getVolume({
-        after: timestampYesterday,
+        after: getTimestampYesterday(),
         txnType,
         accounts: this.accountPath,
       })
     )?.outgoingSats ?? 0
 
-  return outgoingSats + amount > limit
+  return threshold - outgoingSats
+}
+
+UserSchema.methods.remainingWithdrawalLimit = async function () {
+  if (!this.oldEnoughForWithdrawal) return 0
+
+  const withdrawalLimit = yamlConfig.limits["withdrawal"].level[this.level]
+
+  const outgoingSats =
+    (
+      await User.getVolume({
+        after: getTimestampYesterday(),
+        txnType: [{ type: { $ne: "on_us" } }],
+        accounts: this.accountPath,
+      })
+    )?.outgoingSats ?? 0
```

Comment:
same remarks here on the repetition of `?.outgoingSats ?? 0`

Reply:
Fixed